src/formElementPlugins/number/src/ApplicationView.tsx

```diff
@@ -0,0 +1,169 @@
+    // Ensures defaultValue is saved on first load
+    if (!currentResponse?.text && defaultValue !== undefined) {
+      const [number, text] = parseInput(
+        String(defaultValue),
+        numberFormatter,
+        simple,
+        prefix,
+        suffix
+      )
+      const validation = customValidate(number, type, minValue, maxValue)
+      setInternalValidation(validation)
+      if (validation.isValid) onSave({ text, number, type, currency, locale })
+      setTextValue(text)
+    }
+  }, [])
+
+  useEffect(() => {
+    if (!textValue) return
+    handleLoseFocus()
+  }, [locale, minValue, maxValue, currency, prefix, suffix, maxSignificantDigits, type])
+
+  function handleChange(e: any) {
+    const text = e.target.value
+    const [number, _] = parseInput(text, numberFormatter, simple, prefix, suffix)
+    setInternalValidation(customValidate(number, type, minValue, maxValue))
+    onUpdate(text)
+    setTextValue(text)
+  }
+
+  function handleLoseFocus() {
+    const [number, text] = parseInput(textValue, numberFormatter, simple, prefix, suffix)
+    const validation = customValidate(number, type, minValue, maxValue)
+    setInternalValidation(validation)
+    if (validation.isValid) onSave({ text, number, type, currency, locale })
+    else onSave(null)
+    setTextValue(text)
+  }
+
+  const styles = maxWidth
+    ? {
+        maxWidth,
+      }
+    : {}
+
+  return (
+    <>
+      <label>
+        <Markdown text={label} semanticComponent="noParagraph" />
+      </label>
+      <Markdown text={description} />
+      <Form.Input
+        fluid
+        type={simple ? 'number' : 'text'}
+        min={minValue}
+        max={maxValue}
+        step={step}
+        placeholder={placeholder}
+        onChange={handleChange}
+        onBlur={handleLoseFocus}
+        onFocus={setIsActive}
+        value={textValue ? textValue : ''}
+        disabled={!isEditable}
+        style={styles}
+        error={
+          !validationState.isValid || !internalValidation.isValid
+            ? {
+                content:
+                  internalValidation?.validationMessage || validationState?.validationMessage,
+                pointing: 'above',
+              }
+            : null
+        }
+      />
+    </>
+  )
+}
+
+const parseInput = (
+  textInput: string | null | undefined,
+  numberFormatter: any,
+  simple: boolean,
+  prefix: string,
+  suffix: string
+): [number | null, string | null] => {
+  if (!textInput) return [null, null]
+  const number: number = Number(textInput?.replace(/[^\d\.\-]/g, ''))
+  const text: string = simple
+    ? String(number)
+    : ((prefix + numberFormatter.format(number) + suffix) as string)
+  return [number, text]
+}
+
+const customValidate = (
+  number: number | null | undefined,
+  type: NumberType,
+  minValue: number = -Infinity,
+  maxValue: number = Infinity
+): { isValid: boolean; validationMessage?: string } => {
+  if (number === NaN || number === null) return { isValid: true }
```

Comment:
```suggestion
  if (number === NaN || number === null) return { isValid: false }
```

? Not sure why NAN or Null would return a true validation, I'm confused

Reply:
Yeah, good point. I was thinking that "NULL" responses are valid by the normal rules of our plugins, but I've updated it now to give more useful error state when a non-number is entered. 37c27c1d

Thanks